### ISSUE TITLE
Pulumi convert command defaults to destructive action

### ISSUE THREAD

**Issue by catmeme** [pulumi/pulumi#12842]
### What happened?

The default behavior of `pulumi convert` without specifying the `--out` flag is to overwrite the following files in the current directory:

```
.gitignore
index.ts
package.json
Pulumi.yaml
tsconfig.json
```

### Expected Behavior

A warning with prompt to continue, or a default `out` directory

### Steps to reproduce

To reproduce, run `pulumi convert` without the `--out` flag, in an existing project.  (git used to illustrate the deleted content).

```
git init
pulumi new static-website-aws-yaml
git add .
pulumi convert --language typescript
git diff
```

### Output of `pulumi about`

```
CLI
Version      3.66.0
Go Version   go1.20.4
Go Compiler  gc

Plugins
NAME  VERSION
yaml  unknown
```

### Contributing

Vote on this issue by adding a 👍 reaction. 
To contribute a fix for this issue, leave a comment (and link to your pull request, if you've opened one already).

**Comment by justinvp** [comment]
Yes, that's a great point. We should fix it along the lines you suggest. Thanks for opening the issue!

**Comment by Frassle** [comment]
Duplicate of https://github.com/pulumi/pulumi/issues/9891
